module/VuFind/src/VuFind/Controller/AlmaController.php

```diff
@@ -0,0 +1,342 @@
+				}
+		
+                return $this->webhookUser($requestBodyJson);
+                break;
+            case 'JOB_END':
+            case 'NOTIFICATION':
+            case 'LOAN':
+            case 'REQUEST':
+            case 'BIB':
+            case 'ITEM':
+                return $this->webhookNotImplemented($webhookAction);
+            	break;
+            default:
+            	$accessPermission = 'access.alma.webhook.challenge';
+            	try {
+					$this->checkPermission($accessPermission);
+				} catch (\VuFind\Exception\Forbidden $ex) {
+					return $this->createJsonResponse('Access to Alma Webhook challenge forbidden. Set permission \''.$accessPermission.'\' in \'permissions.ini\'.', 403);
+				}
+                return $this->webhookChallenge();
+                break;
+        }  
+    }
+    
+    /**
+     * Webhook actions related to a newly created, updated or deleted user in Alma.
+     * 
+     * @param mixed $requestBodyJson		A JSON string decode with json_decode()
+     * @return NULL|\Zend\Http\Response
+     */
+    protected function webhookUser($requestBodyJson) {
+    	
+    	// Initialize user variable that should hold the user table row
+        $user = null;
+        
+        // Initialize response variable
+        $jsonResponse = null;
+        
+        // Get method from webhook (e. g. "create" for "new user")
+        $method = (isset($requestBodyJson->webhook_user->method)) ? $requestBodyJson->webhook_user->method : null;
+        
+        // Get primary ID
+        $primaryId = (isset($requestBodyJson->webhook_user->user->primary_id)) ? $requestBodyJson->webhook_user->user->primary_id : null;
+        
+        if ($method == 'CREATE' || $method == 'UPDATE') {
+            // Get username (could e. g. be the barcode)
+            $username = null;
+            $userIdentifiers = (isset($requestBodyJson->webhook_user->user->user_identifier)) ? $requestBodyJson->webhook_user->user->user_identifier : null;
+            $idTypeConfig = ($this->configAlma->NewUser->idType != null && isset($this->configAlma->NewUser->idType)) ? $this->configAlma->NewUser->idType : null;
+            foreach ($userIdentifiers as $userIdentifier) {
+                $idTypeHook = (isset($userIdentifier->id_type->value)) ? $userIdentifier->id_type->value : null;
+                if ($idTypeHook != null && $idTypeHook == $idTypeConfig && $username == null) {
+                    $username = (isset($userIdentifier->value)) ? $userIdentifier->value : null;
+                }
+            }
+            
+            // Use primary ID as username as a fallback if no other username ID is available
+            $username = ($username == null) ? $primaryId : $username;
+            
+            // Get user details from Alma Webhook message
+            $firstname = (isset($requestBodyJson->webhook_user->user->first_name)) ? $requestBodyJson->webhook_user->user->first_name : null;
```

Comment:
Now that we've switched to requiring PHP 7, you can take advantage of the `??` operator to simplify all of these isset checks -- i.e. rather than `isset($foo) ? $foo : null` you can write `$foo ?? null`. It greatly improves readability of code like this!